### ISSUE TITLE
Question about performance on A100

### ISSUE THREAD

**Issue by a-r-r-o-w** [thu-ml/SageAttention#3]
Hey :wave: This is amazing work, thanks!

I'm trying to use the example code on an A100 but don't notice any speedups. Is this expected? I do see a 20-30% speedup on a 4090, so maybe it's due to differences between the GPUs.

I'm using the following code from your example to test:

```python
import torch
from diffusers import CogVideoXPipeline
from diffusers.utils import export_to_video
from sageattention import sageattn
import torch.nn.functional as F

F.scaled_dot_product_attention = sageattn

prompt = "A panda, dressed in a small, red jacket and a tiny hat, sits on a wooden stool in a serene bamboo forest. The panda's fluffy paws strum a miniature acoustic guitar, producing soft, melodic tunes. Nearby, a few other pandas gather, watching curiously and some clapping in rhythm. Sunlight filters through the tall bamboo, casting a gentle glow on the scene. The panda's face is expressive, showing concentration and joy as it plays. The background includes a small, flowing stream and vibrant green foliage, enhancing the peaceful and magical atmosphere of this unique musical performance."

pipe = CogVideoXPipeline.from_pretrained(
    "THUDM/CogVideoX-2b",
    torch_dtype=torch.float16
).to("cuda")

video = pipe(
    prompt=prompt,
    num_videos_per_prompt=1,
    num_inference_steps=50,
    num_frames=49,
    guidance_scale=6,
    generator=torch.Generator(device="cuda").manual_seed(42),
).frames[0]

export_to_video(video, "output.mp4", fps=8)
```

**Comment by jt-zhang** [comment]
Thank you for your kind words and for reaching out 😊.

Currently, sageattention is optimized for the 4090 and 3090 GPUs. Optimization for the A100 will be available soon.

**Comment by a-r-r-o-w** [comment]
Thanks again for the awesome work, good to know there's a plan to support additional kernels for other GPUs and I look forward to it! Will try to make improvements on my end as well for A100/H100s

**Comment by LSC527** [comment]
I also run a test on A100, sageattn is slower than torch sdpa on A100. Looking forward to optimization on A100.

**Comment by jason-huang03** [comment]
We will provide optimized kernels for A100 in the the upcoming release. However, theoretical acceleration on A100 is at most $(1 + 1) / (0.5 + 1) = 1.333$, not counting for cuda core overhead like softmax. As such, this kernel serves primarily for benchmarking rather than for productive purpose.

**Comment by liangan1** [comment]
> We will provide an optimized kernel for A100 in the the upcoming release. However, theoretical acceleration on A100 is at most ( 1 + 1 ) / ( 0.5 + 1 ) = 1.333 , not counting for cuda core overhead like softmax. As such, this kernel serves primarily for benchmarking rather than for productive purpose.

Why the 4090 has different projection speedup?

**Comment by jason-huang03** [comment]
Please refer to [ada white paper](https://images.nvidia.com/aem-dam/Solutions/Data-Center/l4/nvidia-ada-gpu-architecture-whitepaper-v2.1.pdf) for tensor core throughput on 4090, you can find the theoratical speedup on 4090 is higher.

**Comment by Zhaojp-Frank** [comment]
> Please refer to [ada white paper](https://images.nvidia.com/aem-dam/Solutions/Data-Center/l4/nvidia-ada-gpu-architecture-whitepaper-v2.1.pdf) for tensor core throughput on 4090, you can find the theoratical speedup on 4090 is higher.

I found **some conflicts** btw your paper and NVIDIA Ada white paper (WP) about 4090, i.e, 
1. P6 of paper said “Matmul using INT8 is **two times** faster than using FP8 in many commonly used GPUs, e.g., RTX4090.”, however based on P30 of WP, their peak perf is exact the same at 660 TFLops on 4090.
2. In fact, according to the WP, FP8 (with FP16 accu), FP8 (with FP32 acc) and INT8 all three format offers the same peak perf at 660. 
would u double check the number? thanks.
![image](https://github.com/user-attachments/assets/28dc2637-23f1-40ba-89b3-37b57e36c3c2)

**Comment by jason-huang03** [comment]
This version of white paper has error. The later updated white paper says fp8 with fp32 accumulate is 330T. Check [here](https://images.nvidia.cn/aem-dam/Solutions/geforce/ada/nvidia-ada-gpu-architecture.pdf).
